### PR TITLE
Add "chain completed/progressed/failed" text to mission debrief

### LINKED ISSUE #379: Evaluate using "global effects" to say "chain is advanced"
image

Could be an interesting place to put it

### PR DESCRIPTION
**Pull request by cannonfodder1**
Closes #379

### CODE REVIEW

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc [hunk 1]
```diff
@@ -840,6 +840,31 @@ static function BarracksStatusReport GetBarracksStatusReport()
...
+{
+	local XComGameState_ActivityChain ChainState;
+	local string ChainName;
+
+	ChainState = ActivityState.GetActivityChain();
+	ChainName = ChainState.GetMyTemplate().strTitle;
+
+	if (bVictory)
+	{
+		if (ChainState.GetLastActivity().GetMyTemplate() == ActivityState.GetMyTemplate())
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]:
```uc
if (ChainState.GetLastActivity().ObjectID == ActivityState.ObjectID)
```

File: CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc [hunk 2]
```diff
@@ -840,6 +840,31 @@ static function BarracksStatusReport GetBarracksStatusReport()
...
+			return "'" $ ChainName $ "' Chain Completed";
+		}
+		else
+		{
+			return "'" $ ChainName $ "' Chain Progressed";
+		}
+	}
+	else
+	{
+		return "'" $ ChainName $ "' Chain Failed";
```
Comment (Xymanek) [inline CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: That isn't a necessity though.... It all depends on what `Template.ShouldProgressChain()` returns.

In fact, the chain framework itself has no concept of a "failed chain" - only one with a `eACER_ProgressBlocked` - although that's how it's currently used....

IMO we should add some vars to the activity template or just ignore this "issue" altogether and just leave this version (with assumptions).

In any case, **localization**

Comment (cannonfodder1) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: If the chain is not supposed to end on a failed activity, then the activity template can specify a special OnFailure function that displays a different message.

Comment (Xymanek) [reply CovertInfiltration/Src/CovertInfiltration/Classes/X2Helper_Infiltration.uc]: My dreams of chain-agnostic activities shattered -.-

**Pull request merged**